protocol/messenger.go

```diff
@@ -869,6 +870,26 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 		return nil, err
 	}
 
+	displayName, err := m.settings.DisplayName()
+	if err != nil {
+		return nil, err
+	}
+	if err := utils.ValidateDisplayName(&displayName); err != nil {
+		// Somehow a wrong display name was saved. We need to update it so that others accept our messages
+		pubKey, err := m.settings.GetPublicKey()
+		if err == nil {
+			replacementDisplayName := pubKey[:12]
+			m.logger.Warn("unaccepted display name was saved to the setting, reverting to pubkey substring", zap.String("displayName", displayName), zap.String("replacement", replacementDisplayName))
+
+			if err := m.SetDisplayName(replacementDisplayName); err != nil {
```

Comment:
This is highly unexpected to be an error, right? I would return `err` up here.

Reply:
This one could return an error since it does some extra validations and even checks for duplicates. So to be extra safe, I think I won't return the error